yt/frontends/rockstar/io.py

```diff
@@ -99,8 +99,13 @@ def _initialize_index(self, data_file, regions):
             pos = np.empty((halos.size, 3), dtype="float64")
             # These positions are in Mpc, *not* "code" units
             pos = data_file.ds.arr(pos, "code_length")
-            dx = np.finfo(halos['particle_position_x'].dtype).eps
-            dx = 2.0*self.ds.quan(dx, "code_length")
+            eps = np.finfo(halos['particle_position_x'].dtype).eps
+            # Make sure eps is not larger than the domain itself
+            if(eps>np.max(abs(self.ds.domain_right_edge))):
```

Comment:
I think this is actually a safer strategy in all cases.  For example, if `domain_right_edge` is greater than 2, the current calculation will not work.  I suggest removing the if statement and making this the sole method.

Reply:
Okay, I removed it.